pkg/services/searchV2/auth.go

```diff
@@ -6,22 +6,22 @@ import (
 	"github.com/grafana/grafana/pkg/infra/db"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/dashboards"
 	"github.com/grafana/grafana/pkg/services/sqlstore/permissions"
-	"github.com/grafana/grafana/pkg/services/sqlstore/searchstore"
 	"github.com/grafana/grafana/pkg/services/user"
 )
 
 // ResourceFilter checks if a given a uid (resource identifier) check if we have the requested permission
-type ResourceFilter func(uid string) bool
+type ResourceFilter func(kind entityKind, uid, parent string) bool
```

Comment:
```suggestion
type ResourceFilter func(kind entityKind, uid, parentUID string) bool
```

Reply:
also, once we have the nested folders - will this still work or will we have to refactor this interface?

Reply:
We will most likely have to tackle that separately. However, the old implementation would not have worked for nested folders either, as we have to do a bunch of work to make RBAC compatible with nested folders in general.